execute/plugin.go

```diff
@@ -172,6 +166,9 @@ func (p *Plugin) Observation(
 		}
 	}
 
+	msgVisibilityInterval := p.cfg.MessageVisibilityInterval
+	fetchFrom := time.Now().Add(-msgVisibilityInterval)
```

Comment:
I'm not 100% sure about this but does this time need to be in UTC? We're comparing this to block timestamps eventually right?